### ISSUE TITLE
Defining multiple rules for container app

### ISSUE THREAD

**Issue by Seifeldin-Sabry** [microsoft/azure-container-apps#1040]
### This issue is a: 
- [ ] bug report -> please search issues before submitting
- [X] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
I am trying to run an update containerapp command passing in a yaml file. In that yaml file I define multiple rules (1 HTTP, 2 servicebus).

I am defining the rules as you see below. however I tried another variation like this (but this didn't work either, and I'm aware its missing the HTTP rule):
```yaml
properties:
  scale:
    minReplicas: 0
    maxReplicas: 1
    rules:
      - name: azure-clean-data-queue-rule
        custom:
          type: azure-servicebus
          metadata:
            queueName: clean-data-cute
            namespace: ${AZURE_SERVICEBUS_NAMESPACE}
            messageCount: 1000
      - name: azure-prediction-queue-rule
        custom:
          type: azure-servicebus
          metadata:
            queueName: prediction-cute
            namespace: ${AZURE_SERVICEBUS_NAMESPACE}
            messageCount: 1000
```

### Steps to reproduce
1. create a quick container app
2. run container app update and pass this yaml file:
```yml
properties:
  scale:
    minReplicas: 0
    maxReplicas: 1
    rules:
      - custom:
          auth:
            - secretRef: "azure-sb-connection"
              triggerParameter: connection
          metadata:
            messageCount: 1000
            namespace: "secretref:azure-sb-namespace"
            queueName: "prediction-cute"
          type: azure-servicebus
        name: azure-prediction-queue-scale-rule
      - custom:
          auth:
            - secretRef: "azure-sb-connection"
              triggerParameter: connection
          metadata:
            messageCount: 1000
            namespace: "secretref:azure-sb-namespace"
            queueName: "clean-data-cute"
          type: azure-servicebus
        name: azure-clean-data-queue-scale-rule
      - http:
          name: http-scale-rule
          metadata:
            concurrentRequests: '10'
```
EDIT: updated the yaml file as best as I could. I converted from json to yaml from the docs. and copied it. It doesn't just see the last rule. it isn't clear which one it picks but it just picks one

**Expected behavior** [What you expected to happen.]
I expect 3 rules to show up in the azure portal

**Actual behavior** [What actually happened.]
only the last rule is shown

### Additional context

This issue occurred in CLI, its a deploy CI script this runs, worked fine before we needed multiple rules, so we didn't use a yaml file.

**Comment by anthonychu** [comment]
The YAML has a couple of issues:

- `scale` is a property inside of the template (`properties.template`)
- In the `http` scale rule, `name` is a sibling of `http`, not a child
- All metadata must be strings, so `messageCount` values should be `'1000'`

This is the updated YAML that you can try:

```yaml
properties:
  template:
    scale:
      minReplicas: 0
      maxReplicas: 1
      rules:
        - custom:
            auth:
              - secretRef: "azure-sb-connection"
                triggerParameter: connection
            metadata:
              messageCount: '1000'
              namespace: "secretref:azure-sb-namespace"
              queueName: "prediction-cute"
            type: azure-servicebus
          name: azure-prediction-queue-scale-rule
        - custom:
            auth:
              - secretRef: "azure-sb-connection"
                triggerParameter: connection
            metadata:
              messageCount: '1000'
              namespace: "secretref:azure-sb-namespace"
              queueName: "clean-data-cute"
            type: azure-servicebus
          name: azure-clean-data-queue-scale-rule
        - http:
            metadata:
              concurrentRequests: '10'
          name: http-scale-rule
```

You might find it a bit simpler to export YAML from the app and edit it:
`az containerapp show -n my-app -g my-group -o yaml > my-app.yaml`

Sorry for the difficulties you're experiencing. We're investigating how we can improve the YAML experience, such as returning actionable errors.

**Comment by Seifeldin-Sabry** [comment]
Oh great! thank you, errors would definetly help. I realized something wasn't right because I set the rules manually and then rerun the script to try to override it. But since it didn't even reset to an empty array I was pretty confused.

I tried many different documentation, but since I was filtering it to only what I needed, I might've overlooked the `template` property. Thanks a lot for your time!
